rfc/system/0000-exteranl-plugin-service.md

```diff
@@ -0,0 +1,270 @@
+# Out of Core Plugin [RFC]
+
+## Motivation
+Currently it is hard to implement backend plugins, especially for data-scientists & MLE’s who do not have working knowledge of Golang. Also, performance requirements, maintenance and development is cumbersome.
+
+The document here proposes a path to make it possible to write plugins rapidly, while decoupling them from the core flytepropeller engine.
+
+## Goals
+* Plugins should be easy to author - no need of code generation, using tools that MLEs and Data Scientists are not accustomed to using.
+* Most important plugins for Flyte today are plugins that communicate with external services.
+* It should be possible to test these plugins independently and also deploy them privately.
+* It should be possible for users to use backend plugins for local development, especially in flytekit and unionML
+* It should be possible to author plugins in any language.
+* Plugins should be scalable
+* Plugins should have a very simple API
+* Plugins should show up in the UI (extra details)
+
+## Overview
+**Flytekit Backend Plugin** is a plugin registry written by fastAPI. Both users and Propeller can send the request to it to run the jobs (Bigquery, Databricks). Moreover, this registry should be stateless, so we can easily to scale up to multiple machines.
+
+![](https://i.imgur.com/3FQcW99.png)
+
+![](https://i.imgur.com/9xBeEri.jpg)
+
+
+
+**FlytePropeller Backend System Plugin**: New web api plugin used to send the request to flytekit backend service to create a job on an external platform.
+
+### Create Job
+```sequence
+Propeller->Grpc API Server: REST (Task spec)
```

Comment:
This suggests that propeller will talk to the gRPC api server via REST? Is this correct? The paragraph below confirms that users should be able to write plugins using a "rest-like interface". Why do we call it gRPC API server here then?